### PR TITLE
Fix OPENVINO_TF_DISABLE env var check and HDDL backend name checks

### PR DESCRIPTION
**Pull request by ck-intel**

### CODE REVIEW

**Comment by kannanx** [comment]
Verified the fixes on the PR. The fixes are found to be working as expected.

File: openvino_tensorflow/grappler/ovtf_optimizer.cc [hunk 1]
```diff
@@ -62,8 +62,16 @@ Status OVTFOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
   // OPENVINO_TF_DISABLE is
   // set
   // we will not do anything; all subsequent passes become a no-op.
-  bool ovtf_not_enabled =
-      (!api::IsEnabled()) || (std::getenv("OPENVINO_TF_DISABLE") != nullptr);
+  bool ovtf_not_enabled = false;
+  const char* opnevino_tf_disable_env = std::getenv("OPENVINO_TF_DISABLE");
```
Comment (sspintel) [inline openvino_tensorflow/grappler/ovtf_optimizer.cc:66]: openvino is misspelled

File: openvino_tensorflow/rewrite_pass.cc [hunk 2]
```diff
@@ -109,8 +109,16 @@ class NGraphEncapsulationPass : public NGraphRewritePass {
     // is set
     // we will not do anything; all subsequent
     // passes become a no-op.
-    bool ovtf_not_enabled =
-        (!api::IsEnabled()) || (std::getenv("OPENVINO_TF_DISABLE") != nullptr);
+    bool ovtf_not_enabled = false;
+    const char* opnevino_tf_disable_env = std::getenv("OPENVINO_TF_DISABLE");
+    if (!(opnevino_tf_disable_env == nullptr)) {
```
Comment (sspintel) [inline openvino_tensorflow/rewrite_pass.cc:114]: openvino is misspelled

File: openvino_tensorflow/grappler/ovtf_optimizer.cc (hunk 1 above)
Comment (ck-intel) [reply openvino_tensorflow/grappler/ovtf_optimizer.cc]: Done

File: openvino_tensorflow/rewrite_pass.cc (hunk 2 above)
Comment (ck-intel) [reply openvino_tensorflow/rewrite_pass.cc]: done

**Pull request merged**